lib/puppet/face/config.rb

```diff
@@ -94,16 +96,47 @@
 
         values = Puppet.settings.values(Puppet[:environment].to_sym, options[:section].to_sym)
         report_section_and_environment(options[:section], Puppet.settings[:environment])
-        if args.length == 1
-          puts values.interpolate(args[0].to_sym)
-        else
-          args.sort.each do |setting_name|
-            puts "#{setting_name} = #{values.interpolate(setting_name.to_sym)}"
-          end
+
+        to_be_rendered = {}
+        args.sort.each do |setting_name|
+          to_be_rendered[setting_name] = values.interpolate(setting_name.to_sym)
         end
       end
-      nil
+
+      # convert symbols to strings before formatting output
+      if render_all_settings
+        to_be_rendered = stringifyhash(to_be_rendered)
+      end
+      to_be_rendered
+    end
+
+    when_rendering :console do |to_be_rendered|
+      output = ''
+      if to_be_rendered.keys.length > 1
+        to_be_rendered.keys.sort.each do |setting|
+          output << "#{setting} = #{to_be_rendered[setting]}\n"
+        end
+      else
+        output << "#{to_be_rendered.to_a[0].last}\n"
```

Comment:
Can `to_be_rendered` be an empty hash:

```
{}.to_a[0].last
NoMethodError: undefined method `last' for nil:NilClass
```

Reply:
There are 3 cases:
* no args are given - all settings are printed
* a valid setting is specified - you get the setting key value pair
* an invalid setting is specified - you get the setting name and nil value